frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/services/MfaResetIPVAuthorizationService.java

```diff
@@ -0,0 +1,200 @@
+package uk.gov.di.authentication.frontendapi.services;
+
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jose.jwk.RSAKey;
+import com.nimbusds.jwt.EncryptedJWT;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+import com.nimbusds.oauth2.sdk.AuthorizationRequest;
+import com.nimbusds.oauth2.sdk.ResponseType;
+import com.nimbusds.oauth2.sdk.id.ClientID;
+import com.nimbusds.oauth2.sdk.id.State;
+import com.nimbusds.oauth2.sdk.id.Subject;
+import com.nimbusds.oauth2.sdk.token.AccessToken;
+import com.nimbusds.openid.connect.sdk.OIDCClaimsRequest;
+import com.nimbusds.openid.connect.sdk.claims.ClaimsSetRequest;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import uk.gov.di.audit.AuditContext;
+import uk.gov.di.authentication.shared.entity.ResponseHeaders;
+import uk.gov.di.authentication.shared.entity.Session;
+import uk.gov.di.authentication.shared.helpers.IdGenerator;
+import uk.gov.di.authentication.shared.helpers.NowHelper.NowClock;
+import uk.gov.di.authentication.shared.serialization.Json;
+import uk.gov.di.authentication.shared.services.AuditService;
+import uk.gov.di.authentication.shared.services.CloudwatchMetricsService;
+import uk.gov.di.authentication.shared.services.ConfigurationService;
+import uk.gov.di.authentication.shared.services.RedisConnectionService;
+import uk.gov.di.authentication.shared.services.SerializationService;
+import uk.gov.di.authentication.shared.services.TokenService;
+
+import java.security.interfaces.RSAPublicKey;
+import java.time.Clock;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+import static uk.gov.di.authentication.frontendapi.domain.FrontendAuditableEvent.REVERIFY_AUTHORISATION_REQUESTED;
+import static uk.gov.di.authentication.shared.helpers.ApiGatewayResponseHelper.generateApiGatewayProxyResponse;
+
+public class MfaResetIPVAuthorizationService {
+    private static final Logger LOG = LogManager.getLogger(MfaResetIPVAuthorizationService.class);
+    private static final JWSAlgorithm SIGNING_ALGORITHM = JWSAlgorithm.ES256;
+    private static final String MFA_RESET_SCOPE = "reverification";
+    private static final String STATE_STORAGE_PREFIX = "state:";
```

Comment:
We're going to want a different prefix for this I think, that is used elsewhere

Reply:
Have updated to `mfaReset:state`